image/image.go

```diff
@@ -86,3 +86,40 @@ func (image *Image) RootFS(ctx context.Context, provider content.Provider) ([]di
 
 	return diffIDs, nil
 }
+
+// Size returns the total size of an image's packed resources.
+func (image *Image) Size(ctx context.Context, provider content.Provider) (int64, error) {
+	var size int64
+	return size, Walk(ctx, HandlerFunc(func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
+		switch image.Descriptor.MediaType {
+		case MediaTypeDockerSchema2Manifest, ocispec.MediaTypeImageManifest:
+			size += desc.Size
+			rc, err := provider.Reader(ctx, image.Descriptor.Digest)
+			if err != nil {
+				return nil, err
+			}
+			defer rc.Close()
+
+			p, err := ioutil.ReadAll(rc)
```

Comment:
`p` is shorter for `pointer`. It is used commonly, throughout Go, to denote a local byte slice.

Either way, the context here is limited, so this acceptable.